src/web/crate_details.rs

```diff
@@ -587,4 +619,112 @@ mod tests {
+            "authors": [],
+            "owners": [],
+            "authors_json": null,
+            "dependencies": null,
+            "release_time": super::super::duration_to_str(time),
+            "build_status": true,
+            "last_successful_build": null,
+            "rustdoc_status": true,
+            "repository_url": null,
+            "homepage_url": null,
+            "keywords": null,
+            "have_examples": true,
+            "target_name": "x86_64-unknown-linux-gnu",
+            "releases": [],
+            "github": true,
+            "yanked": false,
+            "github_stars": null,
+            "github_forks": null,
+            "github_issues": null,
+            "metadata": {
+                "name": "serde",
+                "version": "1.0.0",
+                "description": "serde does stuff",
+                "target_name": null,
+                "rustdoc_status": true,
+                "default_target": "x86_64-unknown-linux-gnu"
+            },
+            "is_library": true,
+            "doc_targets": [],
+            "license": null,
+            "documentation_url": null
+        });
+
+        assert_eq!(correct_json, serde_json::to_value(&details).unwrap());
+
+        details.description = Some("serde does stuff".to_string());
+        details.owners = vec![("Owner".to_string(), "owner@ownsstuff.com".to_string())];
+
+        let authors = vec![("Somebody".to_string(), "somebody@somebody.com".to_string())];
+        details.authors_json = Some(serde_json::to_value(&authors).unwrap());
+        details.authors = authors;
+
+        let correct_json = json!({
+            "name": "rcc",
+            "version": "100.0.0",
+            "description": "serde does stuff",
+            "authors": [["Somebody", "somebody@somebody.com"]],
+            "owners": [["Owner", "owner@ownsstuff.com"]],
+            "authors_json": [["Somebody", "somebody@somebody.com"]],
+            "dependencies": null,
+            "release_time": super::super::duration_to_str(time),
+            "build_status": true,
+            "last_successful_build": null,
+            "rustdoc_status": true,
+            "repository_url": null,
+            "homepage_url": null,
+            "keywords": null,
+            "have_examples": true,
+            "target_name": "x86_64-unknown-linux-gnu",
+            "releases": [],
+            "github": true,
+            "yanked": false,
+            "github_stars": null,
+            "github_forks": null,
+            "github_issues": null,
+            "metadata": {
+                "name": "serde",
+                "version": "1.0.0",
+                "description": "serde does stuff",
+                "target_name": null,
+                "rustdoc_status": true,
+                "default_target": "x86_64-unknown-linux-gnu"
+            },
+            "is_library": true,
+            "doc_targets": [],
+            "license": null,
+            "documentation_url": null
+        });
```

Comment:
All this duplicate code makes me really uncomfortable, it also makes it really hard to read the diffs (both the commit diff and the changes from the previous `correct_json`). Does serde_json have some sort of record update syntax, the way rust has `struct S { field1: x, ..other_s }`?